x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/EsQueryExec.java

```diff
@@ -94,6 +95,18 @@ public SortBuilder<?> sortBuilder() {
         }
     }
 
+    public record ScoreSort(Order.OrderDirection direction) implements Sort {
+        @Override
+        public SortBuilder<?> sortBuilder() {
+            return new ScoreSortBuilder();
+        }
+
+        @Override
+        public FieldAttribute field() {
+            return null;
```

Comment:
I think `Sort` would need a refactor, this isn't quite right like this. We should add a TODO and maybe an entry into an issue to track it.

Reply:
ok, I've added a TODO and a task in #116599 meta-issue